src/geode/inspector/criterion/intersections/surface_intersections.cpp

```diff
@@ -0,0 +1,404 @@
+                    }
+                    const auto t1_edge_inter_pt_id =
+                        vertex_position_to_index( edge_edge_inter.first );
+                    const auto t2_edge_inter_pt_id =
+                        vertex_position_to_index( edge_edge_inter.second );
+                    if( t1_edge_inter_pt_id == geode::NO_LID
+                        || t2_edge_inter_pt_id == geode::NO_LID )
+                    {
+                        return true;
+                    }
+                    if( edge1_vertices[t1_edge_inter_pt_id]
+                            != edge2_vertices[t2_edge_inter_pt_id]
+                        || edge1_vertices[t1_edge_inter_pt_id]
+                               != common_points[0] )
+                    {
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    bool triangle_intersects_other( const geode::Triangle3D& t1,
+        const geode::Triangle3D& t2,
+        const geode::PolygonVertices& t1_vertices,
+        const geode::PolygonVertices& t2_vertices,
+        absl::Span< const geode::index_t > common_points )
+    {
+        for( const auto v_id : geode::LRange{ 3 } )
+        {
+            const auto v2_id = v_id == 2 ? 0 : v_id + 1;
+            const auto intersection = segment_triangle_intersection_detection(
+                { t1.vertices()[v_id], t1.vertices()[v2_id] }, t2 );
+            if( intersection.first != geode::Position::outside )
+            {
+                if( common_points.size() != 1 )
+                {
+                    return true;
+                }
+                const auto edge_inter_pt_id =
+                    vertex_position_to_index( intersection.first );
+                const auto t2_inter_pt_id =
+                    vertex_position_to_index( intersection.second );
+                if( edge_inter_pt_id == geode::NO_LID
+                    || t2_inter_pt_id == geode::NO_LID )
+                {
+                    return true;
+                }
+                if( t1_vertices[edge_inter_pt_id == 0 ? v_id : v2_id]
+                        != t2_vertices[t2_inter_pt_id]
+                    || t2_vertices[t2_inter_pt_id] != common_points[0] )
+                {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    template <>
+    bool TriangleTriangleIntersection< 3 >::triangles_intersect(
+        geode::index_t t1_id,
+        geode::index_t t2_id,
+        absl::Span< const geode::index_t > common_points ) const
+    {
+        const auto t1_vertices = mesh_.polygon_vertices( t1_id );
+        const auto t2 = mesh_.triangle( t2_id );
+        if( common_points.size() == 2 )
+        {
+            const auto& t1_third_pt =
+                mesh_.point( third_point_index( t1_vertices, common_points ) );
+            if( geode::segment_triangle_intersection_detection(
```

Comment:
Why not use seg_seg instead of seg_tri? or why is this code different in 2D?

Reply:
seg_seg in 3D does not exists I suppose